### PR TITLE
Remove api version dependency for hide key board extension

### PR DESCRIPTION
**Pull request by hugokallstrom**
### Checklist

- [x] Functionality is covered by an integration test
- [ ] Functionality is accessible in engineering mode

### CODE REVIEW

File: app/src/main/java/com/hedvig/app/util/extensions/ContextExtensions.kt [hunk 1]
```diff
@@ -78,9 +78,13 @@ fun Context.hideKeyboard(view: View) {
     inputMethodManager.hideSoftInputFromWindow(view.windowToken, 0)
 }
 
-@RequiresApi(Build.VERSION_CODES.R)
-suspend fun Context.hideKeyboardIfVisible(view: View, inputView: View, delayMillis: Long = 0) {
-    if (view.rootWindowInsets?.isVisible(WindowInsets.Type.ime()) == true) {
+suspend fun Context.hideKeyboardWithDelay(inputView: View, delayMillis: Long = 0) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+        if (inputView.rootView.rootWindowInsets?.isVisible(WindowInsets.Type.ime()) == true) {
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/util/extensions/ContextExtensions.kt:83]: I think there's actually a compat-variant of this API! We should check that out, in that case we could move to only having one code path

**Review by hedvigoscar — APPROVED** [review]
Looks good enough to merge, really, but I would like it if we investigated the Compat-approach 🙂

File: app/src/main/java/com/hedvig/app/util/extensions/ContextExtensions.kt (hunk 1 above)
Comment (hugokallstrom) [reply app/src/main/java/com/hedvig/app/util/extensions/ContextExtensions.kt:83]: Investigating!

Comment (hugokallstrom) [reply app/src/main/java/com/hedvig/app/util/extensions/ContextExtensions.kt:83]: We need `androidx.core:core-ktx:1.5.0-beta03` In order to access `WindowInsetsCompat.Type`. Should we upgrade or wait for stable?

Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/util/extensions/ContextExtensions.kt:83]: I'm ok with upgrading 🙂

Comment (hugokallstrom) [reply app/src/main/java/com/hedvig/app/util/extensions/ContextExtensions.kt]: Done!

File: app/src/main/java/com/hedvig/app/util/extensions/ContextExtensions.kt [hunk 2]
```diff
@@ -78,9 +79,8 @@ fun Context.hideKeyboard(view: View) {
     inputMethodManager.hideSoftInputFromWindow(view.windowToken, 0)
 }
 
-@RequiresApi(Build.VERSION_CODES.R)
-suspend fun Context.hideKeyboardIfVisible(view: View, inputView: View, delayMillis: Long = 0) {
-    if (view.rootWindowInsets?.isVisible(WindowInsets.Type.ime()) == true) {
+suspend fun Context.hideKeyboardWithDelay(inputView: View, delayMillis: Long = 0) {
+    if (inputView.rootView.rootWindowInsets?.isVisible(WindowInsetsCompat.Type.ime()) == true) {
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/util/extensions/ContextExtensions.kt:83]: This needs to use the compat-equivalents, or it will break 😞

Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/util/extensions/ContextExtensions.kt:83]: i.e the `rootWindowInsets`-compat ones

Comment (hugokallstrom) [reply app/src/main/java/com/hedvig/app/util/extensions/ContextExtensions.kt:83]: Ah good catch, fixed

**Pull request merged**